test/property.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+var test = require('tape')
+var fc = require('fast-check')
+var _ = require('lodash')
+var is = require('..')
+
+test('unist-util-is properties', function (t) {
+  t.plan(4)
+  t.doesNotThrow(function () {
+    fc.assert(
+      fc.property(
+        fc.record({type: fc.string({minLength: 1})}),
+        function (node) {
+          return is(node)
+        }
+      )
+    )
+  }, 'should check if given node (#1)')
+
+  t.doesNotThrow(function () {
+    fc.assert(
+      fc.property(
+        fc.dictionary(fc.string(), fc.string()).filter(function (node) {
+          return typeof node.type === 'undefined'
+        }),
+        function (node) {
+          return !is(node)
+        }
+      )
+    )
+  }, 'should check if given node (#2)')
+
+  t.doesNotThrow(function () {
+    fc.assert(
+      fc.property(
+        fc.record({type: fc.string({minLength: 1})}),
+        function (node) {
+          return is(node, node.type)
+        }
+      )
+    )
+  }, 'should match types')
+
+  t.doesNotThrow(function () {
+    fc.assert(
+      fc.property(
+        fc
+          .unicodeJsonObject()
+          // Filter for JSON objects which unist can work with
+          .filter(function (node) {
+            return (
+              // JSON needs to be a plain object
+              _.isPlainObject(node) &&
+              // Also needs to have some keys with primitive values
+              _.some(_.keys(node), function (key) {
+                return !_.isObject(node[key])
+              })
+            )
+          })
+          // Return node and a list with a random subset of it's primitive value keys
+          .chain(function (node) {
+            return fc.tuple(
+              fc.constant(node),
+              fc.subarray(
+                _.filter(_.keys(node), function (key) {
+                  return !_.isObject(node[key])
+                }),
+                {minLength: 1}
+              )
+            )
+          }),
+        fc.string({minLength: 1}),
+        function (nodeAndKeys, type) {
+          var nodeProperties = nodeAndKeys[0]
+          var keys = nodeAndKeys[1]
+
+          var node = _.assign(nodeProperties, {type: type})
+          var subsetOfNode = _.pick(_.cloneDeep(node), keys)
+          return is(node, subsetOfNode)
```

Comment:
there's a fair amount of lodash here to avoid using ES6 and newer JS methods.
If ES6, `Array.filter`, `Array.some`, and `Object.assign` can be used this can be cleaned up

Reply:
given that fast-check uses es2017, test code here is fine using that version too IMO

Reply:
updated to use ES2017 features